prov/util/src/util_attr.c

```diff
@@ -162,8 +162,19 @@ static int ofi_info_to_core(uint32_t version, const struct fi_provider *prov,
 	const char *core_name;
 	size_t len;
 
-	if (!(*core_hints = fi_allocinfo()))
-		return -FI_ENOMEM;
+	/* If we are here through fi_fabric we must allocate only fabric_attr */
+	if (util_info && util_info->fabric_attr && util_info->fabric_attr->name &&
+	    !util_info->domain_attr) {
```

Comment:
This function (ofi_info_to_core) allocates a new fi_info for core_hints (to be passed to core provider) and initializes needed fields based on util_hints (passed from util provider). If we got here from fi_fabric, only fabric_attr has to be allocated and initialized. fi_allocinfo by default allocates memory for domain_attr. This would result in mr_mode check failing in underlying provider (as none of the mr_mode bits would be set in core_hints).